test/mitmproxy/tools/console/test_master.py

```diff
@@ -0,0 +1,30 @@
+from unittest.mock import Mock
+
+
+def test_spawn_editor(monkeypatch, console):
+    text_data = "text"
+    binary_data = b"\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09"
+
+    console.get_editor = Mock()
+    console.get_editor.return_value = "editor"
+    console.get_hex_editor = Mock()
+    console.get_hex_editor.return_value = "editor"
+    monkeypatch.setattr("subprocess.call", (lambda _: None))
```

Comment:
I dislike that we need to do so much mocking here, but that's mostly a fault of the existing code. If you feel this could be improved I'd be open for a refactor, but also ok if we want to keep it that way.

Reply:
I was having trouble trying to understand how the console worked, so I ended up just mocking it, to not leave to PR hanging. I can try to come up with a more appropriate solution for that. I actually wanted to create more tests, including some unrelated to this change, that I thought was missing.